html5ever/src/tokenizer/mod.rs

```diff
@@ -1565,7 +1578,10 @@ impl<Sink: TokenSink> Tokenizer<Sink> {
 
             states::MarkupDeclarationOpen => go!(self: error; to BogusComment),
 
-            states::CdataSection => go!(self: emit_temp; error_eof; to Data),
+            states::CdataSection => {
+                self.bad_eof_error();
```

Comment:
This reorders the eof and emit_tmp. Is that a problem?

Reply:
Any implementation that relies on this ordering is doing silly things, but it is observable. Nice catch!